### PR TITLE
Remove deprecated APIs

### PR DESCRIPTION
**Pull request by kevinsawicki**
This pull request queues up the removal of the deprecated APIs that will **not** be included in the **1.0** release.

It does leave the `deprecate` and `deprecations` helper classes for post 1.0 usage when more APIs need to be deprecated.

/cc @electron/maintainers

### CODE REVIEW

**Comment by deepak1556** [comment]
`lib/browser/api/protocol.js` has some api to be deprecated.

File: lib/browser/api/app.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
 'use strict'
 
-const electron = require('electron')
-const {deprecate, session, Menu} = electron
+const {Menu} = require('electron')
 const EventEmitter = require('events').EventEmitter
 
+// Initialize session
+require('electron').session
```
Comment (deepak1556) [inline lib/browser/api/app.js]: why initializing here ?

Comment (kevinsawicki) [inline lib/browser/api/app.js]: It crashes without this require.

Comment (deepak1556) [inline lib/browser/api/app.js]: have pushed a commit that should fix the crash.

**Comment by kevinsawicki** [comment]
> lib/browser/api/protocol.js has some deprecated api to be removed.

@deepak1556 awesome, thanks so much 👍

**Comment by kevinsawicki** [comment]
@zcbenz Should I retarget this pull request against the `1-0-release` branch?

**Comment by kevinsawicki** [comment]
Retargeted in #5373

**Pull request closed without merging**